### PR TITLE
Add properties to methods

### PR DESCRIPTION
**Pull request by peyerluk**
#### Add format, minLength, maxLength to string()

```js
ms.string({format: 'email', minLength: 3, maxLength: 200})
```

#### Add uniqueItems, minItems, maxItems to arrayOf()

```js
ms.arrayOf(ms.string(), {minItems: 1, maxItems: 3, uniqueItems: true})
```

#### Add title, description, dependencies to obj() and strictObj()

```js
ms.obj({}, {title: 'Title', description: 'Desc.', dependencies: {/* see readme */}})
```

#### Add method const()

```js
const ms = require('microschema')
ms.const('foobar')
```

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -178,3 +199,12 @@ function isString (str) {
...
   return typeof nr === 'number'
 }
+
+function setDependencies (jsonSchema, dependencies) {
+  const formattedDeps = {}
+  for (const prop in dependencies) {
+    const value = dependencies[prop]
+    formattedDeps[prop] = isString(value) ? [value] : value
+  }
+  jsonSchema.dependencies = formattedDeps
```
Comment (marcbachmann) [inline index.js]: Oh wow, I didn't know this is possible now.
And `ajv` also already supports it. ❤️

File: README.md [hunk 2]
```diff
@@ -180,6 +204,22 @@ output = {
...
+  displayName: 'string',
+}, {title: 'Title', description: 'Desc.'})
+```
+
+Add dependencies:
+```js
+ms.obj({
+  creditCard: 'string',
+  address: 'string'
+}, {dependencies: {creditCard: 'address'}})
```
Comment (marcbachmann) [inline README.md]: What's your use case for the usage of `dependencies`?
Those are weird examples 😁 

I'd like to have a dependency when a specific property has a certain value, which still won't work. The rest isn't that interesting.

Comment (peyerluk) [reply README.md]: I mainly wanted to add `format` to string. But added a bunch more while I was at it. We can leave `dependencies` out of this PR if you prefer. I don't have a use case at the moment.

The example is from over here: https://spacetelescope.github.io/understanding-json-schema/reference/type.html

Comment (peyerluk) [reply README.md]: For the use case with a specifc property value I think we have to use `oneOf()` with `const()`. Or is there a better alternative?

```js
{
  oneOf: [ms.obj({
    type: ms.const('a'),
    foo: 'string'
  }), ms.obj({
    type: ms.const('b'),
    bar: 'string'
  })]
}
```

Comment (marcbachmann) [reply README.md]: no, if you want to cover it with a schema, then there is not.
In our project I've manually handled it using code to get a better error message: https://github.com/upfrontIO/livingdocs-server/blob/master/app/features/menus/menu_api.js#L252-L259

Comment (peyerluk) [reply README.md]: Shall I merge?

**Pull request merged**